providers/standard/src/airflow/providers/standard/sensors/date_time.py

```diff
@@ -99,9 +99,6 @@ def poke(self, context: Context) -> bool:
 
     @property
     def _moment(self) -> datetime.datetime:
-        if isinstance(self.target_time, datetime.datetime):
-            return self.target_time
-
```

Comment:
Rationale: Removed as the two documented options in constructor are converted to `str` during init(), so in this part of code no datetime can land.

Reply:
FAIL: This conversion handling is needed because if Templating is used and the fields are `render_template_as_native_obj=True` then the rendering is generating a native datetime object --> Need to revert this and ignore warning.

Reply:
Reverted. and added comment.